Dockerfile.train.tmpl

```diff
@@ -1,5 +1,9 @@
 # Please refer to the TRAINING documentation, "Basic Dockerfile for training"
 
+# Pull in CUDA so that the GPU can be accessed from within the Docker container 
+FROM nvidia/cuda:10.2-base
```

Comment:
Not only this is not needed because of https://github.com/tensorflow/tensorflow/blob/r1.15/tensorflow/tools/ci_build/Dockerfile.gpu#L1 but it's also wrong since TensorFlow depens on 10.0

Reply:
I've tested this several times on two different machines. 

If I include the lines in the `Dockerfile` as above, spawning the Docker container with the command: 

```
$ sudo docker run  -it \
  --entrypoint /bin/bash \
  --gpus all \
  --mount type=volume,source=deepspeech-data,target=/DeepSpeech/persistent-data c2ec2476fedb
```

works correctly, and the Docker container is spawned, and training using `DeepSpeech.py` occurs using the GPU on the machine. 

If the lines are _not_ included in the `Dockerfile`, spawning the container results in: 

```
docker: Error response from daemon: could not select device driver "" with capabilities: [[gpu]].
ERRO[0000] error waiting for container: context canceled 
```

Both machines are running Ubuntu 20.04 LTS. One is a System 76 Oryx Pro 3, with a GTX 1060 and the other is a Dell Precision 5500, running a Quadro M1200. 

Very happy to investigate further.

Reply:
Have you installed nvidia-docker ? You are not passsing `--runtime=nvidia`

Reply:
> sudo docker

that's also not a good sign, you should be able to run `docker` without specifc credentials. This could be a symptom of something improperly configured on your system.

Reply:
It could also be this weird nvidia issue that we circumvent here: https://github.com/common-voice/commonvoice-fr/blob/5a0f61baf112620286b30319eb7000c57d8a20d0/DeepSpeech/checks.sh#L7-L8

Reply:
FTR, even on Github they advise to use the TensorFlow image directly: https://github.com/NVIDIA/nvidia-docker/issues/1414#issuecomment-725963382

Reply:
I'm also having issues running NVIDIA-Docker on Debian/Sid at the moment, that seems to be broken (cgroups-related?).

Reply:
@KathyReid GPUs are properly visible from the container after:
 - bypassing cgroup for the recent aforementioned regression regarding cgroups (it was running a few weeks ago, a bug against nvidia-docker github has been filed for debian testing)
 - passing devices manually on command-line (side-effect of bypassing cgroups): ` --device /dev/nvidia0 --device /dev/nvidiactl --device /dev/nvidia-modeset --device /dev/nvidia-uvm --device /dev/nvidia-uvm-tools`
 - running `ldconfig`

```
$ docker run --device /dev/nvidia0 --device /dev/nvidiactl --device /dev/nvidia-modeset --device /dev/nvidia-uvm --device /dev/nvidia-uvm-tools --runtime=nvidia -it tensorflow/tensorflow:1.15.4-gpu-py3 /bin/bash


________                               _______________                
___  __/__________________________________  ____/__  /________      __
__  /  _  _ \_  __ \_  ___/  __ \_  ___/_  /_   __  /_  __ \_ | /| / /
_  /   /  __/  / / /(__  )/ /_/ /  /   _  __/   _  / / /_/ /_ |/ |/ / 
/_/    \___//_/ /_//____/ \____//_/    /_/      /_/  \____/____/|__/


WARNING: You are running this container as root, which can cause new files in
mounted volumes to be created as the root user on your host machine.

To avoid this, run the container by specifying your user's userid:

$ docker run -u $(id -u):$(id -g) args...

root@f090a6cd79d5:/# 
root@f090a6cd79d5:/# ldconfig
^[[3~root@f090a6cd79d5:/# dmesg 
dmesg: read kernel buffer failed: Operation not permitted
root@f090a6cd79d5:/# nvidia-smi 
Sun Jan 10 14:58:15 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.45.01    Driver Version: 455.45.01    CUDA Version: 11.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 208...  On   | 00000000:21:00.0  On |                  N/A |
| 25%   49C    P8    21W / 250W |    244MiB / 11016MiB |      1%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
+-----------------------------------------------------------------------------+
```

Reply:
```
$ docker run --device /dev/nvidia0 --device /dev/nvidia1 --device /dev/nvidiactl --device /dev/nvidia-modeset --device /dev/nvidia-uvm --device /dev/nvidia-uvm-tools --runtime=nvidia -it tensorflow/tensorflow:1.15.4-gpu-py3 /bin/bash

________                               _______________                
___  __/__________________________________  ____/__  /________      __
__  /  _  _ \_  __ \_  ___/  __ \_  ___/_  /_   __  /_  __ \_ | /| / /
_  /   /  __/  / / /(__  )/ /_/ /  /   _  __/   _  / / /_/ /_ |/ |/ / 
/_/    \___//_/ /_//____/ \____//_/    /_/      /_/  \____/____/|__/


WARNING: You are running this container as root, which can cause new files in
mounted volumes to be created as the root user on your host machine.

To avoid this, run the container by specifying your user's userid:

$ docker run -u $(id -u):$(id -g) args...

root@73c63b431cc0:/# nvidia-smi 
NVIDIA-SMI couldn't find libnvidia-ml.so library in your system. Please make sure that the NVIDIA Display Driver is properly installed and present in your system.
Please also try adding directory that contains libnvidia-ml.so to your system PATH.
root@73c63b431cc0:/# ldconfig
root@73c63b431cc0:/# nvidia-smi 
Sun Jan 10 15:02:20 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.45.01    Driver Version: 455.45.01    CUDA Version: 11.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 208...  On   | 00000000:21:00.0  On |                  N/A |
|  0%   50C    P8    22W / 250W |    244MiB / 11016MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 208...  On   | 00000000:4B:00.0 Off |                  N/A |
|  0%   41C    P8    22W / 250W |      6MiB / 11019MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
+-----------------------------------------------------------------------------+